jsrc/algorithm.hpp

```diff
@@ -21,19 +37,108 @@ zip_find(P pred, I f, I l, Is... fs) {
  * @brief Predicate version of `algo::zip_found`
  */
 template <typename P, typename I, typename... Is>
-[[nodiscard]] auto
-zip_found(P pred, I f, I l, Is... fs) {
+[[nodiscard]] static constexpr auto
+zip_found(P const &pred, I f, I const l, Is... fs) -> bool {
     auto const t = zip_find(pred, f, l, fs...);
     return std::get<0>(t) != l;
 }
 
+// This 2 arg function takes priority so will not work when called from constexpr functions. This isn't an issue in
+// cpp20.
+// /**
+//  * @brief Predicate version of `std::mismatch`
+//  * @note std::mismatch isn't constexpr till cpp20
+//  */
+// template <typename I, typename I2>
+// [[nodiscard]] static auto
+// is_mismatched(I f, I const l, I2 f2) -> bool {
+//     return std::mismatch(f, l, f2).first != l;
+// }
+
+/**
+ * @brief Variadic, Predicate version of `std::mismatch`
+ */
+template <typename I, typename... Is>
+[[nodiscard]] static constexpr auto
+is_mismatched(I f, I const l, Is... fs) -> bool {
+    return zip_found(not_equal_to, f, l, fs...);
+}
```

Comment:
Realized `is_mismatched` is the negation of `std::equal`, so lets just rework this one and remove `is_mismatched`.
```suggestion
is_equal(I f, I const l, Is... fs) -> bool {
    return zip_found(equal_to, f, l, fs...);
}
```